DataFormats/SiStripDetId/interface/SiStripDetId.h

```diff
@@ -112,65 +112,66 @@ class SiStripDetId : public DetId {
 // ---------- inline methods ----------
 
 SiStripDetId::SubDetector SiStripDetId::subDetector() const {
-  if( det() == DetId::Tracker &&
-      subdetId() == static_cast<int>(SiStripDetId::TIB) ) {
-    return SiStripDetId::TIB;
-  } else if ( det() == DetId::Tracker &&
-	      subdetId() == static_cast<int>(SiStripDetId::TID) ) {
-    return SiStripDetId::TID;
-  } else if ( det() == DetId::Tracker &&
-	      subdetId() == static_cast<int>(SiStripDetId::TOB) ) {
-    return SiStripDetId::TOB;
-  } else if ( det() == DetId::Tracker &&
-	      subdetId() == static_cast<int>(SiStripDetId::TEC) ) {
-    return SiStripDetId::TEC;
+  SiStripDetId::SubDetector result;
+  if ( det() == DetId::Tracker) {
+    if ( subdetId() == static_cast<int>(SiStripDetId::TEC) ) {
+      result = SiStripDetId::TEC;
+    } else if ( subdetId() == static_cast<int>(SiStripDetId::TID) ) {
+      result = SiStripDetId::TID;
+    } else if ( subdetId() == static_cast<int>(SiStripDetId::TOB) ) {
+      result = SiStripDetId::TOB;
+    } else if ( subdetId() == static_cast<int>(SiStripDetId::TIB) ) {
+      result = SiStripDetId::TIB;
+    } else {
+      result = SiStripDetId::UNKNOWN;
+    }
   } else {
-    return SiStripDetId::UNKNOWN;
+    result = SiStripDetId::UNKNOWN;
   }
+  return result;
 }
 
 SiStripDetId::ModuleGeometry SiStripDetId::moduleGeometry() const {
+  SiStripDetId::ModuleGeometry geometry;
   switch(subDetector()) {
-  case TIB: return int((id_>>layerStartBit_) & layerMask_)<3? IB1 : IB2;
-  case TOB: return int((id_>>layerStartBit_) & layerMask_)<5? OB2 : OB1;
+  case TIB: geometry = int((id_>>layerStartBit_) & layerMask_)<3? IB1 : IB2;
+  case TOB: geometry = int((id_>>layerStartBit_) & layerMask_)<5? OB2 : OB1;
   case TID: switch ((id_>>ringStartBitTID_) & ringMaskTID_) {
-    case 1: return W1A;
-    case 2: return W2A;
-    case 3: return W3A;
+    case 1: geometry = W1A;
```

Comment:
you are missing "break;" after each assignment (otherwise everything below a selected "case" is executed